### PR TITLE
nvhpc: deploy 23.1 with CUDA 11.8

### PR DESCRIPTION
**Pull request by olupton**
Sits on top of https://github.com/BlueBrain/spack/pull/1606.

### CODE REVIEW

**Comment by olupton** [comment]
Re-running from a clean PR working directory as this has been open for a long time.

**Comment by matz-e** [comment]
> Thank you for your pull request!
>
> Should you want to clear the PR build directory after failures, [please use this pipeline](https://bbpgitlab.epfl.ch/hpc/spack-pr-cleaner/-/pipelines/new?var%5BGITHUB_PULL_REQUEST_ID%5D=1654). Before running the cleanup pipeline, please ensure that any PR building pipelines have been cancelled or finished running.

File: bluebrain/deployment/environments/externals.yaml
Comment (pramodk) [inline bluebrain/deployment/environments/externals.yaml]: @olupton : could you remind me 22.5 was useful or just a buggy for us?

**Review by pramodk — DISMISSED** [review]
LGTM assuming CI passes.

File: bluebrain/deployment/environments/externals.yaml
Comment (olupton) [reply bluebrain/deployment/environments/externals.yaml]: IIRC we never deployed it because 22.7 was available before we resolved https://github.com/BlueBrain/CoreNeuron/issues/826.

I don't have a strong view -- I would mildly prefer to deploy it and then drop it once we have 22.9, as we have more confidence in 22.3 as an "old stable" release but it is occasionally useful to be able to find different versions in the archives.

**Comment by olupton** [comment]
Launched https://bbpgitlab.epfl.ch/hpc/spack-pr-cleaner/-/jobs/494818 to clean up the working directory of this PR.

**Comment by iomaganaris** [comment]
Before deploying this we also need to update the NEURON version with this https://github.com/neuronsimulator/nrn/pull/2239

Edit: done now

**Review by pramodk — APPROVED** [review]
LGTM

@olupton - any other check is required for this?

**Comment by olupton** [comment]
> LGTM
> 
> @olupton - any other check is required for this?

I would say we should launch the NEURON CI against this PR before merging.

**Comment by iomaganaris** [comment]
Launched NEURON CI here: https://bbpgitlab.epfl.ch/hpc/cellular/nrn/-/pipelines/104288

**Review by matz-e — APPROVED** [review]

**Pull request merged**
